djangosaml2/views.py

```diff
@@ -133,6 +135,41 @@ def unknown_idp(self, request, idp):
             msg.format('Please contact technical support.'), status=403
         )
 
+    def load_sso_kwargs_scoping(self, sso_kwargs):
+        """ Performs IdP Scoping if scoping param is present. """
+        idp_scoping_param = self.request.GET.get('scoping', None)
+        if idp_scoping_param:
+            idp_scoping = Scoping()
+            idp_scoping.idp_list = IDPList()
+            idp_scoping.idp_list.idp_entry.append(
+                IDPEntry(provider_id = idp_scoping_param)
+            )
+            sso_kwargs['scoping'] = idp_scoping
+
+    def load_sso_kwargs_authn_context(self, sso_kwargs):
+        # this would work when https://github.com/IdentityPython/pysaml2/pull/807
+        ac = getattr(self.conf, '_sp_requested_authn_context', {})
+
+        # this works even without https://github.com/IdentityPython/pysaml2/pull/807
+        # hopefully to be removed soon !
+        if not ac:
+            scs = getattr(
+                settings, 'SAML_CONFIG', {}
+            ).get('service', {}).get('sp', {})
+            ac = scs.get('requested_authn_context', {})
+        # end transitional things to be removed soon !
+
+        if ac:
+            sso_kwargs["requested_authn_context"] = RequestedAuthnContext(
+                    authn_context_class_ref=[
+                        AuthnContextClassRef(ac['authn_context_class_ref']),
+                    ],
+                    comparison = ac.get('comparison', "minimum"),
+                )
+
+    def load_sso_kwargs(self, sso_kwargs):
+        """ Inherit me if you want to put your desidered things in sso_kwargs """
```

Comment:
This is great. I was thinking of making `sso_kwargs` a class attribute but this is better. I had to make more changes in the Login views to support more configuration attributes but this will simplify it.

Reply:
KISS Is the way 👈